go/obscuronode/enclave/enclave.go

```diff
@@ -124,17 +124,22 @@ func (e *enclaveImpl) Start(block types.Block) {
 }
 
 func (e *enclaveImpl) start(block types.Block) {
-	headerHash := block.Hash()
-	s, f := e.db.FetchState(headerHash)
-	if !f {
-		panic("state should be calculated")
-	}
+	var currentHead *Rollup
+	var currentState RollupState
 
-	currentHead := s.Head
-	currentState := newProcessedState(e.db.FetchRollupState(currentHead.Hash()))
 	var currentProcessedTxs []L2Tx
 	currentProcessedTxsMap := make(map[common.Hash]L2Tx)
 
+	// determine whether the block where the speculative execution will start already contains Obscuro state
+	blockState, f := e.db.FetchState(block.Hash())
+	if f {
+		currentHead = blockState.Head
+		if currentHead != nil {
```

Comment:
@tudor-malene Isn't this non-nil by definition, since if `f` is true then the block is non-nil? `FetchState` is another method where I think it may be simpler to just return the block (and not the boolean), with a nil block indicating "not found". Then you'd only have one check here.

Reply:
ok. let's revisit this when we work on the db. I agree this is not great